### ISSUE TITLE
Hard-coded existsSync call prevents customized TypeScript implementation with alternate file system

### ISSUE THREAD

**Issue by rsms** [ezolenko/rollup-plugin-typescript2#154]
#### What happens and why it is wrong

repro:
- set `typescript: customTypescript` in options
- set `customTypescript` to a version of typescript with a custom `sys` system API where you specifically implement `fileExists` to `console.log(path)` (In reality this probably does something more useful) and `readFile` to return a constant string.
- compile a .ts file importing some non-existing file `import "./does-not-exist.ts"`

expected:
- the "does-not-exist.ts" filename to be printed on stdout at least twice from both `fileExists` and once from `readFile`

actual:
- the custom `fileExists` is only invoked once and `readFile` is never invoked

Example of customized typescript object with alternate system API:

```js
const typescript = require("typescript")
const customTypescript = {
  __proto__: typescript,
  sys: {
    // custom implementation of system API
    __proto__: typescript.sys,
    fileExists(path) {
      console.log("fileExists", path)
      return true
    },
    readFile(path) {
      console.log("readFile", path)
      return ""
    }
  }
}
```

The issue is here:

https://github.com/ezolenko/rollup-plugin-typescript2/blob/fe0490c80755120e988ade29e94e6f36eb08447e/src/host.ts#L50

The fix is probably this:

```diff
- if (existsSync(fileName))
+ if (tsModule.sys.fileExists(fileName))
```

#### Environment 

- nodejs v11.14.0
- macOS 10.14.5 (darwin 18.6.0)

#### Versions

- typescript 3.5.1
- rollup-plugin-typescript2 0.21.1
- rollup 1.14.4
- tslib 1.9.3

#### rollup.config.js

reproduces with any config

#### tsconfig.json

reproduces with any config

**Comment by rsms** [comment]
Note: The implementation of TypeScript already checks for a file's existence, so I think you can simply remove the `if file exists` check from `getScriptSnapshot`.

The call stack looks like this for getScriptSnapshot when TypeScript calls it to load an imported file:

- LanguageServiceHost.getScriptSnapshot
- HostCache.createEntry
- HostCache.getOrCreateEntryByPath
- getOrCreateSourceFileByPath
- getSourceFile
- findSourceFile
- processImportedModules

**Comment by ezolenko** [comment]
Thanks, should be fixed in 0.21.2 now
